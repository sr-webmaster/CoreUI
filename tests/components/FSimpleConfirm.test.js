import { mount } from '@vue/test-utils'
import Component from '../../src/components/FSimpleConfirm.vue'

describe('SimpleConfirm', () => {
  it('snapshot default', async () => {
    const item = {
      title: 'Are you hungry?',
      text: 'Please tell us if you would like a nice hot bowl of mulligatawny soup.  It would really hit the spot!'
    }
    const wrapper = mount(Component, {
      propsData: {
        title: item.title,
        text: item.text
      }
    })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.text()).toContain(item.title)
    expect(wrapper.text()).toContain(item.text)
  })
  test('click on close button', async () => {
    const wrapper = mount(Component)
    wrapper.findAll('button').trigger('click')
    expect(wrapper.emitted().cancel).toHaveLength(1)
    expect(wrapper.emitted().ok).toHaveLength(1)
  })
})
