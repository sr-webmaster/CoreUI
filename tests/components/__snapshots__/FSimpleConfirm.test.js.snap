// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleConfirm snapshot default 1`] = `
<div
  class="simple-confirm pa-3 v-card v-sheet theme--light"
>
  <div
    class="v-card__title px-0 justify-center"
  >
    <div
      class="title font-weight-bold"
    >
      
        Are you hungry?
      
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
   
  <div
    class="py-5 px-4 text-xs-center"
  >
    
      Please tell us if you would like a nice hot bowl of mulligatawny soup.  It would really hit the spot!
    
  </div>
   
  <div
    class="v-card__actions px-8"
  >
    <button
      class="v-btn v-btn--block theme--light light"
      type="button"
    >
      <div
        class="v-btn__content"
      >
        
      Cancel
    
      </div>
    </button>
     
    <div
      class="spacer"
    />
     
    <button
      class="v-btn v-btn--block theme--light primary"
      type="button"
    >
      <div
        class="v-btn__content"
      >
        
      Ok
    
      </div>
    </button>
  </div>
</div>
`;
